From: Dennis Dalessandro <dennis.dalessandro@intel.com>
Subject: IB/qib,rdmavt: Move smi_ah to qib
Git-commit: 4eadd8ff21bffcaf3338267dcca571accc612c8e
Patch-mainline: v4.6
References: bsc#973818, fate#319242
Acked-by: Jiri Bohac <jbohac@suse.cz>

Rdmavt adopted an smi_ah from qib which is not needed by hfi1. Move this
back to qib and get it out of the common library.

Reviewed-by: Ira Weiny <ira.weiny@intel.com>
Reviewed-by: Jubin John <jubin.john@intel.com>
Signed-off-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 drivers/infiniband/hw/qib/qib_iba7322.c |   10 +++++-----
 drivers/infiniband/hw/qib/qib_verbs.h   |    1 +
 include/rdma/rdma_vt.h                  |    1 -
 3 files changed, 6 insertions(+), 6 deletions(-)

--- a/drivers/infiniband/hw/qib/qib_iba7322.c
+++ b/drivers/infiniband/hw/qib/qib_iba7322.c
@@ -2910,8 +2910,8 @@ static void qib_setup_7322_cleanup(struc
 			spin_unlock_irqrestore(&dd->cspec->gpio_lock, flags);
 			qib_qsfp_deinit(&dd->pport[i].cpspec->qsfp_data);
 		}
-		if (dd->pport[i].ibport_data.rvp.smi_ah)
-			ib_destroy_ah(&dd->pport[i].ibport_data.rvp.smi_ah->ibah);
+		if (dd->pport[i].ibport_data.smi_ah)
+			ib_destroy_ah(&dd->pport[i].ibport_data.smi_ah->ibah);
 	}
 }
 
@@ -5507,7 +5507,7 @@ static void try_7322_ipg(struct qib_ppor
 	if (IS_ERR(send_buf))
 		goto retry;
 
-	if (!ibp->rvp.smi_ah) {
+	if (!ibp->smi_ah) {
 		struct ib_ah *ah;
 
 		ah = qib_create_qp0_ah(ibp, be16_to_cpu(IB_LID_PERMISSIVE));
@@ -5515,11 +5515,11 @@ static void try_7322_ipg(struct qib_ppor
 			ret = PTR_ERR(ah);
 		else {
 			send_buf->ah = ah;
-			ibp->rvp.smi_ah = ibah_to_rvtah(ah);
+			ibp->smi_ah = ibah_to_rvtah(ah);
 			ret = 0;
 		}
 	} else {
-		send_buf->ah = &ibp->rvp.smi_ah->ibah;
+		send_buf->ah = &ibp->smi_ah->ibah;
 		ret = 0;
 	}
 
--- a/drivers/infiniband/hw/qib/qib_verbs.h
+++ b/drivers/infiniband/hw/qib/qib_verbs.h
@@ -219,6 +219,7 @@ struct qib_pma_counters {
 
 struct qib_ibport {
 	struct rvt_ibport rvp;
+	struct rvt_ah *smi_ah;
 	__be64 guids[QIB_GUIDS_PER_PORT	- 1];	/* writable GUIDs */
 	struct qib_pma_counters __percpu *pmastats;
 	u64 z_unicast_xmit;     /* starting count for PMA */
--- a/include/rdma/rdma_vt.h
+++ b/include/rdma/rdma_vt.h
@@ -137,7 +137,6 @@ struct rvt_ibport {
 	u16 *pkey_table;
 
 	struct rvt_ah *sm_ah;
-	struct rvt_ah *smi_ah;
 };
 
 #define RVT_CQN_MAX 16 /* maximum length of cq name */
